main.py

```diff
@@ -0,0 +1,420 @@
+            range_start, range_end = request_range.strip().split('-')
+
+            # Try to convert range start/end to byte counts
+            try:
+                self.log_message('range_start: %s', range_start)
+                if range_start:
+                    range_start = int(range_start)
+                self.log_message('range_end: %s', range_end)
+                if range_end:
+                    range_end = int(range_end)
+            except Exception:
+                self.log_message('unsupported range: failed to convert ranges to integers')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Require range start positions
+            if range_start == '':
+                self.log_message('unsupported range: missing range-start')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Do not support range end
+            if range_end != '':
+                self.log_message('unsupported range: range-end was specified')
+                return self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+            # Seek to the beginning of the range in the content
+            f.seek(range_start, SEEK_FROM_BEG)
+
+            # Set the approtriate code/headers for ranged content
+            self.send_response(HTTPStatus.PARTIAL_CONTENT)
+            self.send_header('Content-Range', f'bytes {range_start}-{total_size-1}/{total_size}')
+            self.send_header('Content-Length', str(total_size-range_start))
+        else:
+            # Seek to the beginning of the content since no range was specifieg
+            f.seek(0, SEEK_FROM_BEG)
+
+            # Set the appropriate status code/headers for non-ranged content
+            self.send_response(HTTPStatus.OK)
+            self.send_header('Content-Length', total_size)
+
+        # If a mimetype is known send it to the client
+        if mime_type:
+            self.send_header('Content-Type', mime_type)
+
+        # This header enables http clients to request parts of content, which is important for seeking streaming video
+        self.send_header('Accept-Ranges', 'bytes')
+
+        # This header enables async requests from webpages in browsers enforcing CORS
+        origin = self.headers.get('Origin')
+        if origin:
+            self.send_header('Access-Control-Allow-Origin', origin)
+
+        # No more headers need to be sent
+        self.end_headers()
+
+        if self.method == 'GET':
+            try:
+                shutil.copyfileobj(f, self.wfile)
+            except ConnectionResetError:
+                self.log_error('connection reset by peer %s:%s', self.client_address[0], self.client_address[1])
+
+    def send_movie_genres_index(self):
+	# From https://api.themoviedb.org/3/genre/movie/list?api_key=<<api_key>>&language=en-US
+        genres = [
+            {
+              "id": 28,
+              "name": "Action"
```

Comment:
Should we localize name at some point?

Reply:
https://github.com/duckflix/media-server/issues/5

Reply:
https://github.com/duckflix/media-server/issues/6